samples/subsys/fs/fat_fs/sample.yaml

```diff
@@ -4,4 +4,5 @@ tests:
   sample.filesystem.fat_fs:
     build_only: true
     platform_whitelist: nrf52840_blip olimexino_stm32 mimxrt1050_evk
+        stm32l496g_disco
```

Comment:
I proposed a refactoring of the sample yaml in #24787 to reduce the platform whitelisting. If merged, it means you would instead add `sdhc` to the supported list in your board yaml.

Reply:
Does that mean I should move `DISK_ACCESS_STM32_SDMMC` under `DISK_ACCESS_SDHC` in the Kconfig?

Reply:
That's a good point, I'm not sure if it makes sense to move it. Keeping it there means we'll either need `samples/subsys/fs/fat_fs/prj_stm32l496g_disco.conf` like you originally proposed, or move `CONFIG_DISK_ACCESS_SDHC=y` to board conf overlay files in `samples/subsys/fs/fat_fs/boards/`. I'm ok with either option.

@erwango what do you think?

Reply:
@MaureenHelm I checked and building with `CONFIG_DISK_ACCESS_SDHC=y` does work for the STM32 boards, since all it seems to do is enable the submenu in Kconfig. So it could just stay as is. It's just superfluous for the stm32_sdmmc driver.
Not having the SDHC_SPI enabled is enough to allow it to build. So from that standpoint it would be okay to leave as is.

Reply:
Yes, let's drop the readme.
For the DISK_ACCESS_SDHC, I'm not so sure, I don't think you need it enabled in your case, so it would be part of each board configuration.